src/Preconditions/StringPreconditions.php

```diff
@@ -0,0 +1,105 @@
+<?php namespace Guardsman\Preconditions;
+
+use Guardsman\Exceptions\ArgumentNotAString;
+use Guardsman\Exceptions\StringTooLong;
+use Guardsman\Exceptions\StringTooShort;
+
+trait StringPreconditions
+{
+    /** @return mixed */
+    abstract public function getSubject();
+
+    /**
+     * @throws ArgumentNotAString if the subject is not a string.
+     *
+     * @return $this
+     */
+    public function isString()
```

Comment:
If this is exclusively dealing with multibyte strings maybe it should use [`mb_check_encoding()`](http://php.net/manual/en/function.mb-check-encoding.php).